### PR TITLE
openssl_csr: fix bad tests, avoid accepting invalid crl_distribution_points records

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Nightly CI breaks due to the stricter checks in https://github.com/pyca/cryptography/pull/7710.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
openssl_csr
openssl_csr_pipe

### CODE REVIEW

File: plugins/module_utils/crypto/module_backends/csr.py [hunk 1]
```diff
@@ -289,6 +289,8 @@ def parse_crl_distribution_points(module, crl_distribution_points):
             result.append(cryptography.x509.DistributionPoint(**params))
         except OpenSSLObjectError as e:
             raise OpenSSLObjectError('Error while parsing CRL distribution point #{index}: {error}'.format(index=index, error=e))
+        except ValueError as e:
+            raise OpenSSLObjectError('Error while parsing CRL distribution point #{index}: {error}'.format(index=index, error=e))
```
Comment (kristianheljas) [inline plugins/module_utils/crypto/module_backends/csr.py:293]: Nitpick about duplicated re-raise :)

```suggestion
        except (OpenSSLObjectError, ValueError) as e:
            raise OpenSSLObjectError('Error while parsing CRL distribution point #{index}: {error}'.format(index=index, error=e))
```

**Review by kristianheljas — APPROVED** [review]
Looks good!

File: plugins/module_utils/crypto/module_backends/csr.py [hunk 2]
```diff
@@ -651,7 +653,8 @@ def get_csr_argument_spec():
                         'aa_compromise',
                     ]),
                 ),
-                mutually_exclusive=[('full_name', 'relative_name')]
+                mutually_exclusive=[('full_name', 'relative_name')],
+                required_one_of=[('full_name', 'relative_name', 'crl_issuer')],
```
[the remark this answers is not in the document]
Comment (felixfontein) [reply plugins/module_utils/crypto/module_backends/csr.py:657]: Fixed in 4f8d6d4dcde6dd6b50c9c4899cd7c1337ef28555.

[the remark this answers is not in the document]
Comment (kristianheljas) [reply plugins/module_utils/crypto/module_backends/csr.py:657]: Forgot to mention, `relative_name` and `crl_issuer` suffer the same issue.

**Review by kristianheljas — APPROVED** [review]
Thanks for the improvements @felixfontein, LGTM!

**Comment by felixfontein** [comment]
@kristianheljas thanks for reviewing this!

**Pull request merged**